File: its-my-training-day/training-grounds/leetcode-top-75/questions/trees/tree-dfs/invert-binary-tree/solution.py
import os
import sys
from typing import List, Dict, Optional, Tuple

ROOT_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../../"))
if ROOT_DIR not in sys.path:
    sys.path.append(ROOT_DIR)

from shared.python.ds import ArrayList, ListNode, Stack, Queue, TreeNode, MinHeap, MaxHeap

def invert_binary_tree(root: Optional[TreeNode]) -> Optional[TreeNode]:
    if root is None:
        return None
    left = invert_binary_tree(root.left)
    right = invert_binary_tree(root.right)
    root.left = right
    root.right = left
    return root

File: its-my-training-day/training-grounds/leetcode-top-75/questions/trees/tree-dfs/invert-binary-tree/solution.ts
import {
  ArrayList,
  Stack,
  Queue,
  ListNode,
  TreeNode,
  MinHeap,
  MaxHeap,
} from "../../../../shared/typescript/ds";

export const invertBinaryTree = (root: TreeNode | null): TreeNode | null => {
  if (root === null) return null;
  const left = invertBinaryTree(root.left);
  const right = invertBinaryTree(root.right);
  root.left = right;
  root.right = left;
  return root;
};
